include/xtensor/xassign.hpp

```diff
@@ -359,17 +359,20 @@ namespace xt
         using size_type = typename E1::size_type;
         size_type size = e1.size();
         size_type simd_size = simd_type::size;
+
         size_type align_begin = is_aligned ? 0 : xsimd::get_alignment_offset(e1.raw_data(), size, simd_size);
         size_type align_end = align_begin + ((size - align_begin) & ~(simd_size - 1));
-        for (size_type i = 0; i < align_begin; ++i)
+
+        size_type i = 0;
+        for (; i < align_begin; ++i)
```

Comment:
why this change ? I prefer not to pollute the function scope with loop variable.